File: Soundbox.Reloaded.Ui.Web/scripts/domain/sounds/PlayableSoundDto.ts
module soundbox.domain.sounds {
    export class PlayableSoundDto {
        constructor() {
            this.Sound = new SoundDto();
        }

        public Sound: SoundDto;
        public IsPlaying: boolean;
        public CurrentTime: number;
        public Duration: number;

        public currentTimeToDate(): Date {
            return this.secondsToDate(this.CurrentTime);
        }

        public durationToDate(): Date {
            return this.secondsToDate(this.Duration);
        }

        public currentProgressInPercent(): number {
            return (this.CurrentTime / this.Duration) * 100;
        }

        private secondsToDate(seconds: number): Date {
            var date = new Date(1970, 0, 1);
            date.setSeconds(seconds);
            return date;
        }
    }
}

File: Soundbox.Reloaded.Ui.Presentation/Dto/Sounds/PlayableSoundDto.cs
namespace Soundbox.Reloaded.Ui.Presentation.Dto.Sounds
{
    public class PlayableSoundDto
    {
        public SoundDto Sound { get; set; }

        public bool IsPlaying { get; set; }

        public int CurrentTime { get; set; }

        public int Duration { get; set; }
    }
}
